python/lsst/analysis/drp/calcFunctors.py

```diff
@@ -19,3 +21,44 @@ def setDefaults(self):
         super().setDefaults()
         self.colA.column = "iKronFlux"
         self.colB.column = "iPsFlux"
+
+
+class MagDiff(MultiColumnAction):
+    """Calculate the difference between two magnitudes;
+    each magnitude is derived from a flux column.
+
+    Parameters
+    ----------
+    df : `pandas.core.frame.DataFrame`
+        The catalog to calculate the magnitude difference from.
+
+    Returns
+    -------
+    The magnitude difference in milli mags.
+
+    Notes
+    -----
+    The flux columns need to be in units (specifiable in
+    the fluxUnits1 and 2 config options) that can be converted
+    to janskies. This action doesn't have any calibration
+    information and assumes that the fluxes are already
+    calibrated.
+    """
+
+    col1 = Field(doc="Column to subtract from", dtype=str)
+    fluxUnits1 = Field(doc="Units for col1", dtype=str, default="nanojansky")
+    col2 = Field(doc="Column to subtract", dtype=str)
+    fluxUnits2 = Field(doc="Units for col2", dtype=str, default="nanojansky")
+
+    @property
+    def columns(self):
+        return (self.col1, self.col2)
+
+    def __call__(self, df):
+        flux1 = df[self.col1].values * u.Unit(self.fluxUnits1)
+        mag1 = flux1.to(u.ABmag)
+
+        flux2 = df[self.col2].values * u.Unit(self.fluxUnits2)
+        mag2 = flux2.to(u.ABmag)
+
+        return (mag1 - mag2)*1000.0
```

Comment:
Is multiplying by 1000 always desired here? The rest of the units looks great, thanks!

Reply:
I think so, in all the plots I have made so far it has been. This can always be address in a later ticket if it is needed. Or a different action could be used.